src/sentry/api/endpoints/organization_events.py

```diff
@@ -248,34 +254,32 @@ def _query_non_tag_data(self, key, snuba_args):
                 ['count()', '', 'count'],
                 ['min', 'timestamp', 'first_seen'],
                 ['max', 'timestamp', 'last_seen'],
-                ['uniq', key, 'values_seen'],
             ],
             orderby='-count',
             referrer='api.organization-events-heatmap',
             **snuba_args
         )['data']
         return data
 
-    def _create_tag_key_tag_value_objects(self, key, data):
-        tag_values = []
-        values_seen = 0
-        for datum in data:
-            tag_values.append(TagValue(
-                key=key,
-                value=datum['value'],
-                times_seen=datum['count'],
-                last_seen=datum['last_seen'],
-                first_seen=datum['first_seen'],
-            ))
-            values_seen += datum['count']
-
-        tag_key = TagKey(
-            key=key,
-            values_seen=len(tag_values),
-            count=values_seen,
-            top_values=tag_values,
-        )
-        return tag_key
+    def _create_tag_objects(self, top_values_by_key):
+        tag_keys = []
+        for key, top_values in six.iteritems(top_values_by_key):
+            tag_key = TagKey(key=key, top_values=[], values_seen=len(top_values))
+            total_count = 0
```

Comment:
Total count is now calculated here.

Reply:
It would be good if we can still count a record that has an empty value for that tag as part of the total. In some cases some fields might have a lot of nulls but we still want to include them in % calculation.

Reply:
I also wonder if we need to be going through these tagstore functions to get these values. This function seems like it's mostly designed for getting tags of a specific group which isn't really what we're doing here

Reply:
Gotcha. Changed it. Going to remove the changes to tagstore in a few.